cmd/mockery.go

```diff
@@ -381,6 +381,7 @@ func (r *RootApp) Run() error {
 		InPackage:            r.Config.InPackage,
 		KeepTree:             r.Config.KeepTree,
 		Note:                 r.Config.Note,
+		BuildTags:            r.Config.BuildTags,
```

Comment:
I have (at least) two questions.

1. So, I'm confused about this line. Is this PR attempting to fix this behavior for both `packages` and non-`packages` configurations? Or are you intending to fix it for one or the other? 

2. My other question, is this okay for us to be passing the build tags into `packages.Load`, and to inject those same build tags as a comment into the mock file? My intuition says yes, but I fear there might be a reason why someone would want different tags for the package parsing, and another for the mock file. But I just can't foresee what would be desirable here.

Reply:
1. What is non-packages configuration? Do you mean default top-level configuration?
2. No, I think that is wrong. Because they have different syntax:
* for generated mocks - https://pkg.go.dev/cmd/go#hdr-Build_constraints
* for `packages.Load` - a comma-separated list

Now `tag` config field is used for loading packages but documentation says for generated mocks.

Maybe we need two different config options for generated mocks and package loading?

Reply:
> What is non-packages configuration? Do you mean default top-level configuration?

I mean usage of the legacy configuration semantics, i.e. things that do not use [this](https://vektra.github.io/mockery/latest/features/#packages-configuration).

> Now tag config field is used for loading packages but documentation says for generated mocks. Maybe we need two different config options for generated mocks and package loading?

Yeah, I am thinking this is what we ultimately want to do. Why don't we go ahead and add two new config options:

```
PackagesLoadTags
MockBuildTags
```

I think that should make it clear enough what is intended?

Reply:
To make this more clear, as I had to remind myself too, the current `build-tags` field gets added to the configuration of the [`pkg.Parser` object](https://github.com/vektra/mockery/blob/v2.39.1/pkg/parse.go#L50-L52). This ultimately makes its way to `packages.Load` [here](https://github.com/vektra/mockery/blob/v2.39.1/pkg/parse.go#L73). So `BuildTags` is specifically meant for _only_ `packages.Load`.

What we want, in addition to what we already have, is `mock-build-tags` which is what will ultimately get added as the `//go:build` directive. So this PR is incorrect in the sense that it's sending the `BuildTags` to the `//go:build` string, when it should only go to the `packages.Load` invocation.

Reply:
Sounds good! I intend to add these fields only for the new configuration.

But what is about the current `tags` field? Should I leave it in the source code to be backward compatible and remove from the documentation?

Reply:
I think we should leave it as-is. It's kind of poorly named, but I will make a note of it in the docs clearly explaining what it's used for. In v3 we can probably change the name to make it more obvious.

Reply:
I added the new option and changed derscription of the `tags` option.